numpy/lib/npyio.py

```diff
@@ -791,7 +792,14 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
     """
     # Type conversions for Py3 convenience
     if comments is not None:
-        comments = asbytes(comments)
+        if isinstance(comments, (np.compat.basestring, np.compat.bytes)):
```

Comment:
This is fine. Howeve, both `basestring` and `bytes` are already imported, so the `np.compat` prefix isn't needed.